pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,283 @@
+import functools as ft
+
+import numpy as np
+import pymc3 as pm
+import scipy.stats
+from sklearn.metrics import accuracy_score
+import theano
+
+from pymc3_models.exc import PyMC3ModelsError
+from pymc3_models.models import BayesianModel
+
+
+__all__ = ["GaussianNaiveBayes"]
+
+
+# Gaussian Naive Bayes
+# -----------------------------------------------------------------------------
+
+class GaussianNaiveBayes(BayesianModel):
+    """
+    Naive Bayes classification built using PyMC3.
+
+    The Gaussian Naive Bayes algorithm assumes that the random variables
+    that describe each class and each feature are independent and distributed
+    according to independent Normal distributions.
+
+    See the documentation of the `create_model` method for details on the model
+    itself.
+    """
+
+    def __init__(self):
+        super(GaussianNaiveBayes, self).__init__()
+
+    def create_model(self):
+        """
+        Creates and returns the PyMC3 model.
+
+        We note :math:`x_{jc}` the value of the j-th element of the data vector :math:`x`
+        conditionned on x belonging to the class :math:`c`. The Gaussian Naive Bayes
+        algorithm models :math:`x_{jc}` as:
+
+        .. math::
+
+            x_{jc} \\sim Normal(\\mu_{jc}, \\sigma_{jc})
+
+        While the probability that :math:`x` belongs to the class :math:`c` is given by the
+        categorical distribution:
+
+        .. math::
+
+            P(y=c|x_i) = Cat(\\pi_1, \dots, \\pi_C)
+
+        where :math:`\pi_i` is the probability that a vector belongs to category :math:`i`.
+
+        We assume that the :math:`\pi_i` follow a Dirichlet distribution:
+
+        .. math::
+
+            \\pi \\sim Dirichlet(\\alpha)
+
+        with hyperparameter :math:`\\alpha = [1, .., 1]`. The :math:`\\mu_{jc}`
+        are sampled from a Normal distribution centred on :math:`0` with
+        variance :math:`100`, and the :math:`\\sigma_{jc}` are sampled from a
+        HalfNormal distribuion of variance :math:`100`:
+
+        .. math::
+
+            \\mu_{jc} \\sim Normal(0, 100)
+
+            \\sigma_{jc} \\sim HalfNormal(100)
+
+        Note that the Gaussian Naive Bayes model is equivalent to a Gaussian
+        mixture with a diagonal covariance [1].
+
+        Returns
+        -------
+        A PyMC3 model
+
+        References
+        ----------
+        .. [1] Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+        """
+
+        # The data
+        X = theano.shared(np.zeros((self.num_training_samples, self.num_pred)))
+        y = theano.shared(np.zeros(self.num_training_samples))
+
+        self.shared_vars = {
+            'model_input': X,
+            'model_output': y
+        }
+
+        model = pm.Model()
+        with model:
+            # Priors
+            alpha = np.ones(self.num_cats)
+            pi = pm.Dirichlet('pi', alpha, shape=self.num_cats)
+            mu = pm.Normal('mu', 0, 100, shape=(self.num_cats, self.num_pred))
```

Comment:
Can you explicitly call mu=0 and sd=100 in these distributions? I got bitten when the PyMC3 API switched from tau to sd as default so I prefer to be explicit.